glibcompat.h

```diff
@@ -20,7 +20,7 @@
 #define _GLIBCOMPAT_H_
 
 #if !GLIB_CHECK_VERSION(2, 32, 0)
-#define g_hash_table_contains(hash_table, key) g_hash_table_lookup_extended(hash_table, key, NULL, NULL)
+// #define g_hash_table_contains(hash_table, key) g_hash_table_lookup_extended(hash_table, key, NULL, NULL)

```

Comment:
Any reason to comment this out?  Is `g_hash_table_contains` no longer used in the code?

Reply:
`g_hash_table_contains` is used, but we don't need that macro definition because the `release-2.x.y` branch (the pidgin branch on top of which we build the windows plugin) already has this defined.

Reply:
That's not good :/
That'll cause issues when building against different versions of libpurple.  I'll let grim know.

Linking to `g_hash_table_contains()` in the libpurple.dll makes the resulting libmattermost.dll unportable (as older Pidgin's won't have that function exposed)

If `g_has_table_contains()` is still being used in purple-mattermost then we should continue to use the `#define`

Reply:
if we do use this macro definition, then we will not be able to build for pidgin version >= 2.14.0 because this is already defined in `libpurple/glibcompat.c`. so there are 2 probable solutions.
1. we can document in README that **Minimum Pidgin Version >= 2.14.0** for windows.
2. or, we can build 2 different versions of the plugin, 
     * one for pidgin version < 2.14.0, which includes this macro definition. 
     * one for pidgin version >= 2.14, which does not include this definition.

Reply:
Have talked things over with grim. He's confirmed that the g_hash_table_contains api shouldn't have been made public in 2.14 so is being removed for 2.14.2, and should be gone from the release-2xy branch now 

Does including the macro generate errors? If so, maybe we need to switch to using the lookup-extended check within the code to guarantee forwards and backwards compatibility.

Reply:
yes, it does generate errors. could you provide some reference on what does `lookup-extended check` means? and how can we do that?

Reply:
Sorry, was on phone and typing _ was hard work.  Was meaning to use `g_hash_table_lookup_extended` in the code instead of `g_hash_table_contains` to give max compatibility

Reply:
oh ok, yeah we can do that. will update it

Reply:
Unless you want to rename it to `mm_hash_table_contains` or something more neutral instead?  I don't mind either way :)

Reply:
yeah, `mm_hash_table_contains` seems more reasonable. I have updated the code accordingly.

Reply:
Nice one. Thanks for working this through 👍🏻